### DISCUSSION TITLE
Create a migration script which can run during --apply-migration and move the data around in the db?

### DISCUSSION THREAD

**Discussion by dishankjindal1** [serverpod/serverpod#3023] [Q&A]
I basically created a user table, with some of it fields as non-nullable and now i want to make them nullable. generate migration is throwing a warning. As i am learning the framework, i can face these situations in-future where i want to preserve the data, also want to change types or change field names.

How can i achieve this?

**Comment by hampuslavin** [comment]
Hi @dishankjindal1 !
Making fields nullable should be okay (the opposite, making something non-nullable, can be more work though). Could you please provide the warning your are getting?

**Reply by dishankjindal1** [reply]
Can you share how to write migration, if I am in a situation where I am making my field from null > notnull?

Ofcourse, I am seeking serverpod way of doing it.

**Reply by hampuslavin** [reply]
Given that you have a client as well that you have to be compatible with, I would say this should probably be approached in the same way as renaming a column, which is also a breaking change. There is a good discussion about that here: https://github.com/serverpod/serverpod/discussions/2932#discussioncomment-11142854
